enhancements/network/external-lb-vips.md

```diff
@@ -0,0 +1,213 @@
+4. In the installer, if the platform does support external loadbalancer, some validation will be run
+   to check that the VIPs can already be resolved (the validation already exists for vsphere, we need to
+    make it re-usable). A new field in `PlatformStatus` will be set.
+5. In machine-config-operator, if LB type is external, the internal loadbalancer will not be
+   deployed.
+6. When using an external loadbalancer, the VIPs fields won't be set in `PlatformStatus` and the
+   OpenShift components won't use it when they're not present and LB type is external.
+7. After the deployment, the IP address of a service (e.g. API) can change, as long as the external
+   loadbalancer is updated and the DNS record has been updated and the results are propagated as well.
+
+### API Extensions
+
+A new field will be added to the `PlatformStatus` of the infrastructure object to tell
+the kind of the loadbalancer.
+
+```golang
+// LoadBalancerType can be either external or internal 
+type LoadBalancerType string
+const (
+	// LoadBalancerTypeExternal
+	LoadBalancerTypeExternal LoadBalancerType = "external"
+	// LoadBalancerTypeInternal
+	LoadBalancerTypeInternal LoadBalancerType = "internal"
+)
+// VSpherePlatformStatus holds the current status of the vSphere infrastructure provider.
+type VSpherePlatformStatus struct {
+  // loadBalancerType defines if a internal or external load balancer
+  // is to be used for the openshift api and ingress endpoints.
+  // When set to Internal the static pods defined in machine config operator
+  // will be used.
+  // When set to External the static pods will not be deployed. It will be 
+  // expected that the LB and DNS A records be pre-configured prior to installation.
+  // +kubebuilder:default=internal
+  // +kubebuilder:validation:Enum=internal;external
+  LoadBalancer LoadBalancerType `json:"loadBalancerType,omitempty"`
+}
+```
+
+This field will be set in the `install-config.yaml`:
+```yaml
+platform:
+  vsphere:
+    loadBalancerType: external
+```
+
+### Implementation Details/Notes/Constraints
+
+* CoreDNS will not be deployed if external loadbalancer is being used.
+  The customer is responsible of populating the DNS records for api, api-int and ingress.
+  We need to check who is responsible for machine records.
+
+* We need to make it ready for any platform who wants to support it, and allow
+  platforms to turn this on in the installer project. This will be achieved while validating the
+  installconfig manifest.
+
+* `PlatformStatus` won't have the VIP fields if LB type is external.
+
+
+### Risks and Mitigations
+
+Minimal risk, some platforms (vsphere, openstack) have been testing it already and no issue
+has been seen so far with this scenario.
+Validations will have to be implemented to make sure that the customer has prepare their
+environment before the IPI deployment.
+
+### Drawbacks
+
+This has been asked by customers, so they can have more flexibility so there is nothing to highlight here.
```

Comment:
I mildly disagree. This requires significantly more configuration on the deployer side. That may be a drawback they're accepting, but it's still a drawback.

From our perspective this takes critical infrastructure out of our control. That complicates support of such environments because it means we may need to coordinate with the owner of the external infrastructure to debug problems.

Reply:
ack, I'll add this thought. Thanks